crates/matrix-sdk-crypto/src/session_manager/group_sessions/mod.rs

```diff
@@ -746,9 +756,173 @@ impl GroupSessionManager {
 
         Ok(requests)
     }
+
+    /// Collect the devices belonging to the given user, and send the details of
+    /// a room key bundle to those devices.
+    ///
+    /// Returns a list of to-device requests which must be sent.
+    #[instrument(skip(self, bundle_data))]
+    pub async fn share_room_key_bundle_data(
+        &self,
+        user_id: &UserId,
+        collect_strategy: &CollectStrategy,
+        bundle_data: RoomKeyBundleContent,
+    ) -> OlmResult<Vec<ToDeviceRequest>> {
+        // Only allow conservative sharing strategies
+        let collect_strategy = match collect_strategy {
+            CollectStrategy::AllDevices | CollectStrategy::ErrorOnVerifiedUserProblem => {
+                warn!("Ignoring request to use unsafe sharing strategy {:?} for room key history sharing", collect_strategy);
+                &CollectStrategy::IdentityBasedStrategy
+            }
+            CollectStrategy::IdentityBasedStrategy | CollectStrategy::OnlyTrustedDevices => {
+                collect_strategy
+            }
+        };
+
+        let mut changes = Changes::default();
+
+        let CollectRecipientsResult { devices, .. } =
+            share_strategy::collect_recipients_for_share_strategy(
+                &self.store,
+                iter::once(user_id),
+                collect_strategy,
+                None,
+            )
+            .await?;
+
+        let devices = devices.into_values().flatten().collect();
+        use crate::types::events::EventType;
```

Comment:
Err, can we put this import in a more sensible place? If you don't want to import it in the whole module, importing it at the top of the function seems better to me than in the middle of the function.

Reply:
sorry, yes